app/domain/authentication/authn_gcp/extract_resource_restrictions.rb

```diff
@@ -0,0 +1,104 @@
+require 'command_class'
+
+module Authentication
+  module AuthnGCP
+
+    # This class is responsible of restrictions extraction that are set on a Conjur host or user as annotations.
+    ExtractResourceRestrictions = CommandClass.new(
+      dependencies: {
+        role_class:              ::Role,
+        resource_class:          ::Resource,
+        validate_account_exists: ::Authentication::Security::ValidateAccountExists.new,
+        logger:                  Rails.logger
+      },
+      inputs:       %i(account username extraction_prefix)
+    ) do
+
+      def call
+        validate_account_exists
+        extract_resource_restrictions
+        resource_restrictions
+      end
+
+      private
+
+      def validate_account_exists
+        @validate_account_exists.(
+          account: @account
+        )
+      end
+
+      def extract_resource_restrictions
+        @logger.debug(LogMessages::Authentication::AuthnGCP::ExtractingRestrictionsFromResource.new(@username, @extraction_prefix))
```

Comment:
Couldn't we shorten it using alias or alike, it clutters the code.

Reply:
we decided not to do that.
however, i woulld change the code to:
```
@logger.debug(
  LogMessages::Authentication::AuthnGCP::ExtractingRestrictionsFromResource.new(
    @username, 
    @extraction_prefix
  )
)
```